test/OpenTelemetry.Instrumentation.Grpc.Tests/OpenTelemetry.Instrumentation.Grpc.Tests.csproj

```diff
@@ -45,6 +45,7 @@
   <ItemGroup Condition="'$(RunningDotNetPack)' == 'true'">
     <PackageReference Include="OpenTelemetry.Api" />
     <PackageReference Include="OpenTelemetry.Exporter.InMemory" />
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" />
```

Comment:
@CodeBlanch This change looks bit odd. For my understanding, what part is still pointing to DiagnosticSource 7?

Reply:
Good question. It is kind of interesting. The .NET SDK pulls in a meta reference based on the TFM. Here is an example for net6.0:

![image](https://github.com/open-telemetry/opentelemetry-dotnet/assets/28367120/3678753b-c775-4bf8-83ee-07d6b07c33ae)

^ is showing that what you get by default is the DS version matching the TFM version.

These instrumentation libraries have a really funky build process. When we build them in IDE or in CI they use `ProjectReference`s but when this `RunningDotNetPack` property is `true` they switch to `PackageReference`s (to the current NuGet versions). That is causing all kinds of conflict issues. Probably something we should look at changing!

If we pause to think about this, when using the current NuGets, there really should be no DS 8 at play at all! So the question is really more: What is pulling in DS 8 causing the conflict?!

I just tweaked the fix to be this: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4967/commits/bae46cd0b4b6d5afacb0d2987ee54cfbe6a5fb7d

That takes care of DS 8 coming in so the direct reference is no longer required.